synapse/module_api/__init__.py

```diff
@@ -408,36 +400,39 @@ async def send_local_online_presence_to(self, users: Iterable[str]) -> None:
         Note that this method can only be run on the main or federation_sender worker
         processes.
         """
-        if not self._hs.should_send_federation():
+        local_users = set()
+        remote_users = set()
+        for user in users:
+            if self._hs.is_mine_id(user):
+                local_users.add(user)
+            else:
+                remote_users.add(user)
+
+        if remote_users and not self._hs.should_send_federation():
             raise Exception(
-                "send_local_online_presence_to can only be run "
-                "on processes that send federation",
+                "send_local_online_presence_to can only be called "
+                "with remote users on processes that send federation",
```

Comment:
Heads up that I'm proposing to change `send_local_online_presence_to` to allow calling it on non-federation sender worker processes if you're only calling the method with local users. I think this makes the method more flexible.

Reply:
Although, thinking about it now - this is perhaps a bad smell as your module may break as soon as you start throwing remote users into this method.

I can't really think of anything this would grant you either... So I think I'll just revert this bit.